R/system.R

```diff
@@ -660,26 +660,70 @@ getAmazonAthenaConnection <- function(driver = "", region = "", authenticationTy
   if(Sys.info()["sysname"]=="Linux"){
     driver <-  "/opt/simba/athenaodbc/lib/64/libathenaodbc_sb64.so";
   }
-  # Generate connection string as a key to pool conneciton.
+  # Generate connection string as a key to pool connection.
   connectionString <- stringr::str_c("AwsRegion=",  region, ";AuthenticationType=", authenticationType, ";uid=", user,
                                      ";pwd=", password, ";S3OutputLocation=", s3OutputLocation, ";driver=", driver)
   if(additionalParams != "") {
     connectionString <- stringr::str_c(connectionString,";", additionalParams)
   }
+  if(timezone != "") {
+    connectionString <- stringr::str_c(connectionString,";", timezone)
+  }
+
   conn <- NULL
   if (user_env$pool_connection) {
     conn <- connection_pool[[connectionString]]
   }
   if (is.null(conn)) {
-    conn <- DBI::dbConnect(
-      odbc::odbc(),
-      Driver             = driver,
-      S3OutputLocation   = s3OutputLocation,
-      AwsRegion          = region,
-      AuthenticationType = authenticationType,
-      UID                = user,
-      PWD                = password
-    )
+    if (timezone == "") {
+      timezone <- "UTC" # if timezone is not provided use UTC as default timezone. This is also the default for odbc::dbConnect.
+    }
+    # For Windows, set encoding to make sure non-ascii data is handled properly.
+    # ref: https://github.com/r-dbi/odbc/issues/153
+    if (is.win <- Sys.info()['sysname'] == 'Windows') {
+      loc <- Sys.getlocale(category = "LC_CTYPE")
+      # loc looks like "Japanese_Japan.932", so split it with dot ".".
+      encoding <- stringr::str_split(loc, pattern = "\\.")
+      if (length(encoding[[1]] == 2)) {
+        # encoding looks like: [1] "Japanese_Japan" "932" so check the second part exists or not.
+        conn <- DBI::dbConnect(
+          odbc::odbc(),
+          Driver             = driver,
+          S3OutputLocation   = s3OutputLocation,
+          AwsRegion          = region,
+          AuthenticationType = authenticationType,
+          encoding           = encoding[[1]][[2]],
+          timezone           = timezone,
+          timezone_out       = timezone,
+          UID                = user,
+          PWD                = password
+        )
+      } else {
```

Comment:
Add comment on the difference from the above (encoding).

Reply:
added comment.